File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsBilling.swift
import Foundation

// MARK: - Billing (M9.2)

struct BillingEntitlement: Codable, Identifiable, Hashable {
    var id: String
    var entitlementType: String
    var courseId: String?
    var amountPaidCents: Int
    var subtotalCents: Int?
    var taxAmountCents: Int?
    var taxType: String?
    var taxJurisdiction: String?
    var reverseCharge: Bool?
    var invoiceId: String?
    var currency: String
    var validFrom: String
    var validUntil: String?
    var status: String
}

struct BillingEntitlementsResponse: Codable {
    var entitlements: [BillingEntitlement]?
}

struct BillingTransaction: Codable, Identifiable, Hashable {
    var id: String
    var courseId: String?
    var provider: String
    var providerTxnId: String
    var amountCents: Int
    var currency: String
    var status: String
    var subscriptionId: String?
    var createdAt: String
}

struct BillingTransactionsResponse: Codable {
    var transactions: [BillingTransaction]?
}

struct CheckoutSessionRequest: Encodable {
    var courseId: String?
    var successUrl: String
    var cancelUrl: String
}

struct CheckoutSessionResponse: Decodable {
    var sessionId: String?
    var checkoutUrl: String
    var provider: String?
}

struct CheckoutTaxQuoteRequest: Encodable {
    var courseId: String
}

struct CheckoutTaxQuoteLine: Codable, Hashable {
    var label: String
    var amountCents: Int
}

struct CheckoutTaxQuote: Codable, Hashable {
    var subtotalCents: Int
    var taxAmountCents: Int
    var totalCents: Int
    var currency: String
    var taxType: String?
    var taxJurisdiction: String?
    var reverseCharge: Bool?
    var lines: [CheckoutTaxQuoteLine]
}

struct BillingPortalResponse: Decodable {
    var portalUrl: String
}

struct EntitlementCheckResponse: Decodable {
    var entitled: Bool?
}

struct PendingCheckoutContext: Equatable {
    var courseId: String
    var courseCode: String
    var title: String
}

enum CheckoutReturnPhase: Equatable {
    case success(courseId: String?)
    case cancel
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsBilling.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class BillingEntitlement(
    val id: String,
    val entitlementType: String,
    val courseId: String? = null,
    val amountPaidCents: Int = 0,
    val subtotalCents: Int? = null,
    val taxAmountCents: Int? = null,
    val taxType: String? = null,
    val taxJurisdiction: String? = null,
    val reverseCharge: Boolean? = null,
    val invoiceId: String? = null,
    val currency: String = "USD",
    val validFrom: String,
    val validUntil: String? = null,
    val status: String,
)

@Serializable
data class BillingEntitlementsResponse(
    val entitlements: List<BillingEntitlement>? = null,
)

@Serializable
data class BillingTransaction(
    val id: String,
    val courseId: String? = null,
    val provider: String,
    val providerTxnId: String,
    val amountCents: Int,
    val currency: String,
    val status: String,
    val subscriptionId: String? = null,
    val createdAt: String,
)

@Serializable
data class BillingTransactionsResponse(
    val transactions: List<BillingTransaction>? = null,
)

@Serializable
data class CheckoutSessionRequest(
    val courseId: String? = null,
    val successUrl: String,
    val cancelUrl: String,
)

@Serializable
data class CheckoutSessionResponse(
    val sessionId: String? = null,
    val checkoutUrl: String,
    val provider: String? = null,
)

@Serializable
data class CheckoutTaxQuoteRequest(
    val courseId: String,
)

@Serializable
data class CheckoutTaxQuoteLine(
    val label: String,
    val amountCents: Int,
)

@Serializable
data class CheckoutTaxQuote(
    val subtotalCents: Int,
    val taxAmountCents: Int,
    val totalCents: Int,
    val currency: String,
    val taxType: String? = null,
    val taxJurisdiction: String? = null,
    val reverseCharge: Boolean? = null,
    val lines: List<CheckoutTaxQuoteLine> = emptyList(),
)

@Serializable
data class BillingPortalResponse(
    val portalUrl: String,
)

@Serializable
data class EntitlementCheckResponse(
    val entitled: Boolean? = null,
)

data class PendingCheckoutContext(
    val courseId: String,
    val courseCode: String,
    val title: String,
)

sealed class CheckoutReturnPhase {
    data class Success(val courseId: String?) : CheckoutReturnPhase()
    data object Cancel : CheckoutReturnPhase()
}